go.mod

```diff
@@ -31,11 +31,15 @@ require (
 	google.golang.org/grpc v1.38.0
 	gopkg.in/fsnotify.v1 v1.4.7
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.21.1
-	k8s.io/apimachinery v0.21.1
+	k8s.io/api v0.22.1
+	k8s.io/apimachinery v0.22.1
 	k8s.io/apiserver v0.21.1
 	k8s.io/client-go v0.21.1
-	k8s.io/utils v0.0.0-20210527160623-6fdb442a123b
+	k8s.io/utils v0.0.0-20210707171843-4b05e18ac7d9
 	sigs.k8s.io/controller-runtime v0.9.0
 	sigs.k8s.io/controller-tools v0.4.1
 )
+
+replace k8s.io/client-go => k8s.io/client-go v0.22.0
```

Comment:
Please try without these `replace` directives, and make sure your k8s.io versions are all v0.22.1

Reply:
Tried this but I did not get the latest api changes

Reply:
Yeah, I have done that in the latest commit and it working as expected